consensus/merge/merge.go

```diff
@@ -242,14 +242,25 @@ func (s *Merge) verifyHeader(chain consensus.ChainHeaderReader, header, parent *
 		return consensus.ErrUnexpectedWithdrawals
 	}
 
-	if !chain.Config().IsCancun(header.Time) {
+	cancun := chain.Config().IsCancun(header.Time)
+
+	if cancun && header.ParentBeaconBlockRoot == nil {
+		return fmt.Errorf("missing parentBeaconBlockRoot")
+	}
+	if cancun && header.ParentBeaconBlockRoot != chain.CurrentHeader().ParentBeaconBlockRoot {
+		return fmt.Errorf("parentBeaconBlockRoot mismatch")
+	}
+
+	if !cancun {
 		if header.DataGasUsed != nil {
 			return fmt.Errorf("invalid dataGasUsed before fork: have %v, expected 'nil'", header.DataGasUsed)
 		}
 		if header.ExcessDataGas != nil {
 			return fmt.Errorf("invalid excessDataGas before fork: have %v, expected 'nil'", header.ExcessDataGas)
 		}
-	} else if err := misc.VerifyEip4844Header(chain.Config(), parent, header); err != nil {
+	}
+
+	if err := misc.VerifyEip4844Header(chain.Config(), parent, header); err != nil && cancun {
```

Comment:
Please restore `else if` here. VerifyEip4844Header shouldn't be called for pre-Cancun blocks.